src/modules/margin_tokens/LeveragedToken.ts

```diff
@@ -209,18 +223,55 @@ export default class LeveragedToken extends MarginToken {
     orderData: string,
     options: ContractCallOptions = {},
   ): Promise<object> {
+    const position: Position = await this.margin.getPosition(positionId);
+    const collateral: BigNumber = await this.margin.getPositionBalance(positionId);
+    const principalToClose: BigNumber = this.math.partialAmount(
+      position.principal,
+      collateral,
+      tokensToClose,
+    );
     return this.margin.closePosition(
       positionId,
       closer,
       this.contracts.wethPayoutRecipient.address,
-      tokensToClose,
+      principalToClose,
       ethIsHeldToken,
       exchangeWrapper,
       orderData,
       options,
     );
   }
 
+  public async getPositionOpenedEvents(positionId: string): Promise<any[]> {
```

Comment:
What is this? I don't see it used anywhere and it's not on short token either

I'm not sure why we have these event functions on the leveraged token class. Maybe they could be added to `Margin` if you need them

Reply:
moved over to Margin